File: core/src/main/java/com/lomekwi/cave/timeline/SegmentSetSelectedEvent.java
package com.lomekwi.cave.timeline;

import com.lomekwi.cave.app.selection.SelectableSelectedEvent;
import org.jspecify.annotations.Nullable;

public record SegmentSetSelectedEvent(SegmentSet set, int selectedCount)
    implements SelectableSelectedEvent<SegmentSet> {
    @Override
    public @Nullable SegmentSet selectable() { return set; }
}

File: core/src/main/scala/com/lomekwi/cave/timeline/SegmentSetSelectedEvent.scala
package com.lomekwi.cave.timeline

import com.lomekwi.cave.app.selection.SelectableSelectedEvent

case class SegmentSetSelectedEvent(set: SegmentSet, selectedCount: Int)
    extends SelectableSelectedEvent[SegmentSet] {
  override def selectable(): SegmentSet = set
}
